Error during building solution in Visual Studio

Hello,

Trying to generate a VS 2017 x 64 Solution using CMakeGUI v3.11.1 I got the error:

CMake Error at CMakeLists.txt:23 (add_compile_definitions):
  **Unknown CMake command "add_compile_definitions"**.

When I replaced "add_compile_definitions" through "add_definitions" I could generate a VS solution with 19 projects inside using the CMake Options:

FMI_TYPE CS
FMI_VERSION 3 

But when building the VS solution I got the errors:

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
Severity	Code	Description	Project	File	Line	Suppression State
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	Dahlquist	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	BouncingBall	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	BouncingBall	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	BouncingBall	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	BouncingBall	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	BouncingBall	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	Clocks	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	Clocks	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Clocks	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	Clocks	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Clocks	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	Dahlquist	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Dahlquist	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	Dahlquist	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Dahlquist	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	Feedthrough	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	Feedthrough	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Feedthrough	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	Feedthrough	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Feedthrough	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	LinearTransform	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	LinearTransform	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	LinearTransform	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	LinearTransform	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	LinearTransform	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	Resource	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	Resource	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Resource	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	Resource	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Resource	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	Stair	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	Stair	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Stair	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	Stair	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	Stair	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	VanDerPol	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	VanDerPol	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	VanDerPol	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	VanDerPol	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	VanDerPol	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	co_simulation	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	co_simulation	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	cs_early_return	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	cs_early_return	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	cs_early_return	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be 3	cs_early_return	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	cs_early_return	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	jacobian	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	jacobian	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be 3	jacobian	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	jacobian	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	jacobian	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	model	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	model	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be 3	model	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	model	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	model	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	model_exchange	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	model_exchange	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be 3	model_exchange	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	model_exchange	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	model_exchange	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	scs_synchronous	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	scs_synchronous	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be 3	scs_synchronous	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	scs_synchronous	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	scs_synchronous	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	scs_threaded	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	scs_threaded	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be 3	scs_threaded	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	scs_threaded	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	scs_threaded	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	slave1	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	slave1	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be 3	slave1	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	slave1	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	slave1	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1083	Cannot open source file: '_CRT_SECURE_NO_WARNINGS': No such file or directory	slave2	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1083	Cannot open source file: 'FMI_VERSION=3': No such file or directory	slave2	E:\01_VVI\ModelicaReferenzFMUs\Build\c1	1	
Error	C1189	#error:  FMI_VERSION must be 3	slave2	e:\01_vvi\modelicareferenzfmus\src\fmi3functions.c	9	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	slave2	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
Error	C1189	#error:  FMI_VERSION must be one of 1, 2 or 3	slave2	e:\01_vvi\modelicareferenzfmus\include\model.h	5	
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
could you please provide a solution for this?

Thank you and regards

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.2)
+cmake_minimum_required (VERSION 3.12)
 
 FUNCTION(cat IN_FILE OUT_FILE)
   file(READ ${IN_FILE} CONTENTS)

